cli/tools/upgrade.rs

```diff
@@ -210,30 +210,96 @@ pub fn check_for_upgrades(
-      if version::is_canary() {
-        eprint!(
-          "{} ",
-          colors::green("A new canary release of Deno is available.")
-        );
-        eprintln!(
-          "{}",
-          colors::italic_gray("Run `deno upgrade --canary` to install it.")
-        );
+      if auto_upgrade_env != "1" {
+        if version::is_canary() {
+          eprint!(
+            "{} ",
+            colors::green("A new canary release of Deno is available.")
+          );
+          eprintln!(
+            "{}",
+            colors::italic_gray("Run `deno upgrade --canary` to install it.")
+          );
+        } else {
+          eprint!(
+            "{} {} → {} ",
+            colors::green("A new release of Deno is available:"),
+            colors::cyan(version::deno()),
+            colors::cyan(&upgrade_version)
+          );
+          eprintln!(
+            "{}",
+            colors::italic_gray("Run `deno upgrade` to install it.")
+          );
+        }
+
+        if !auto_upgrade_env.is_empty() {
+          eprintln!(
+            "{} ",
+            colors::yellow(
+              "Auto-upgrade value is set incorrectly, skipping for now."
+            )
+          );
+          eprintln!(
+            "{} ",
+            colors::yellow("To use it, set `DENO_AUTO_UPGRADE=1`")
+          );
+          return;
+        }
+
+        update_checker.store_prompted();
       } else {
-        eprint!(
-          "{} {} → {} ",
-          colors::green("A new release of Deno is available:"),
-          colors::cyan(version::deno()),
-          colors::cyan(&upgrade_version)
-        );
-        eprintln!(
-          "{}",
-          colors::italic_gray("Run `deno upgrade` to install it.")
-        );
-      }
+        if version::is_canary() {
+          eprintln!(
+            "{} ",
+            colors::green("A new canary release of Deno is available.")
+          );
+          eprintln!(
+            "{}",
+            colors::green(
+              "Auto-upgrade enabled, upgrading Deno to latest canary version."
+            )
+          );
+        } else {
+          eprintln!(
+            "{} {} → {} ",
+            colors::green("A new release of Deno is available:"),
+            colors::cyan(version::deno()),
+            colors::cyan(&upgrade_version)
+          );
+          eprintln!(
+            "{}",
+            colors::green(
+              "Auto-upgrade enabled, upgrading Deno to latest version."
+            )
+          );
+        }
+        let upgrade_result = upgrade(
+          Flags::default(),
+          UpgradeFlags {
+            dry_run: false,
+            force: false,
+            canary: version::is_canary(),
+            version: Some(upgrade_version),
+            output: None,
+          },
+        )
+        .await;
 
-      update_checker.store_prompted();
+        match upgrade_result {
+          Ok(_) => {
+            eprintln!("{}", colors::green("Auto-upgrade successful!"));
+          }
```

Comment:
I believe in this case, we should exit the process and inform the user to run the command again - otherwise the upgrade won't take effect until next run.